From 2d6b7bce089e52563bd9c67df62f48e90b48047d Mon Sep 17 00:00:00 2001
From: Julien Thierry <jthierry@redhat.com>
Date: Wed, 6 May 2020 14:30:57 +0100
Subject: [PATCH] Symbol version change

This change causes:
1) Some exported symbols in drivers/base/core.c to see their CRCs
   change.
2) Changes usb_get_dev() referencing a get_device() whose CRC has
   changed, causing the symbol and the new CRC to be included in the
   __version section of the final module.

This makes the final module unloadable for the target kernel.

See "Exported symbol versioning" of the patch author guide for more
detail.

---
diff --git a/drivers/base/core.c b/drivers/base/core.c
index b93f3c5716ae..3bb9dd5d8d74 100644
--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -36,6 +36,8 @@
 #include "physical_location.h"
 #include "power/power.h"
 
+#include <linux/blktrace_api.h>
+
 /* Device links support. */
 static LIST_HEAD(deferred_sync);
 static unsigned int defer_sync_state_count = 1;
diff --git a/drivers/usb/core/usb.c b/drivers/usb/core/usb.c
index a0c432b14b20..da6de67451f0 100644
--- a/drivers/usb/core/usb.c
+++ b/drivers/usb/core/usb.c
@@ -761,6 +761,8 @@ EXPORT_SYMBOL_GPL(usb_alloc_dev);
  */
 struct usb_device *usb_get_dev(struct usb_device *dev)
 {
+	barrier();
+
 	if (dev)
 		get_device(&dev->dev);
 	return dev;
